lib/index.js

```diff
@@ -71,42 +71,50 @@ internals.implementation = (server, options) => {
             const parts = authorization.split(/\s+/);
 
             if (parts[0].toLowerCase() !== settings.tokenType.toLowerCase()) {
-                return reply(settings.unauthorizedFunc(null, settings.tokenType));
+                return settings.unauthorizedFunc(null, settings.tokenType);
             }
 
             const token = parts[1];
 
-            settings.validateFunc.call(request, token, (err, isValid, credentials, artifacts) => {
-
-                if (err) {
-                    return reply(err, { credentials, log: { tags: ['auth', 'bearer'], data: err } });
-                }
+            try {
+                const validationResult = await settings.validateFunc.call(request, token);
```

Comment:
put the try catch only around this call and just declare `validationResult` outside of the try block